relis_app/models/DBConnection_mdl.php

```diff
@@ -770,4 +754,23 @@ function get_list_mdl($ref_table_config, $val = '_', $page = 0, $rec_per_page =
 		return $result;
 
 	}
+
+    /**
+     * Helper function to generate the correct SQL value for a field.
+     * This refactoring was done to eliminate duplicated logic for handling
+     * NULL values and escaping, improving maintainability and readability.
+     *
+     * If the field is a date or an int/number and the value is empty, returns SQL NULL.
+     * Otherwise, returns the value properly escaped for SQL.
+     */
+    private function get_sql_value($value, $val, $db_conn) {
+        if (
+            (isset($value['input_type']) && $value['input_type'] == 'date' && empty($val)) ||
+            (isset($value['field_type']) && in_array($value['field_type'], ['int', 'number']) && empty($val))
+        ) {
+            return "NULL";
```

Comment:
curious why can't we use `return NULL` instead of `return "NULL"` it will be consistent with php null right?